src/transformers/generation/candidate_generator.py

```diff
@@ -111,24 +111,11 @@ def __init__(
         # Prepare the kwargs for the assistant model
         assistant_kwargs = {}
         for key, value in model_kwargs.items():  # deepcopy crashes if we attempt to copy encoder outputs with grads
-            if key not in ("encoder_outputs", "assistant_encoder_outputs"):
+            if key not in ("encoder_outputs", "assistant_encoder_outputs", "past_key_values"):
                 assistant_kwargs[key] = (
                     value.detach().to(device) if isinstance(value, torch.Tensor) else copy.deepcopy(value)
                 )
 
-        # Remove potential default DynamicCache if assistant does not support it
```

Comment:
were tests failing because of this?

Reply:
@ArthurZucker assisted generation if we passed `past_key_values` to `generate` was! The assistant should not copy the cache from the main model by default